pkg/oc/cli/describe/describer.go

```diff
@@ -1202,6 +1203,93 @@ func (d *TemplateDescriber) DescribeTemplate(template *templateapi.Template) (st
+type TemplateInstanceDescriber struct {
+	kubeClient     kclientset.Interface
+	templateClient templateclient.TemplateInterface
+	kprinters.ObjectDescriber
+}
+
+// Describe returns the description of a template instance
+func (d *TemplateInstanceDescriber) Describe(namespace, name string, settings kprinters.DescriberSettings) (string, error) {
+	c := d.templateClient.TemplateInstances(namespace)
+	templateInstance, err := c.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	return d.DescribeTemplateInstance(templateInstance, namespace, settings)
+}
+
+// DescribeTemplateInstance prints out information about the template instance
+func (d *TemplateInstanceDescriber) DescribeTemplateInstance(templateInstance *templateapi.TemplateInstance, namespace string, settings kprinters.DescriberSettings) (string, error) {
+	return tabbedString(func(out *tabwriter.Writer) error {
+		formatMeta(out, templateInstance.ObjectMeta)
+		out.Write([]byte("\n"))
+		out.Flush()
+		d.DescribeConditions(templateInstance.Status.Conditions, out)
+		out.Write([]byte("\n"))
+		out.Flush()
+		d.DescribeObjects(templateInstance.Status.Objects, out)
+		out.Write([]byte("\n"))
+		out.Flush()
+		d.DescribeSecret(namespace, templateInstance.Spec.Secret.Name, out)
+		out.Write([]byte("\n"))
+		out.Flush()
+		return nil
+	})
+}
+
+// DescribeConditions prints out information about the conditions of a template instance
+func (d *TemplateInstanceDescriber) DescribeConditions(conditions []templateapi.TemplateInstanceCondition, out *tabwriter.Writer) {
+	formatString(out, "Conditions", " ")
+	indent := "    "
+	for _, c := range conditions {
+		formatString(out, indent+"Type", c.Type)
+		formatString(out, indent+"Status", c.Status)
+		formatString(out, indent+"LastTransitionTime", c.LastTransitionTime)
+		formatString(out, indent+"Reason", c.Reason)
+		formatString(out, indent+"Message", c.Message)
+		out.Write([]byte("\n"))
+	}
+}
+
+// DescribeObjects prints out information about the objects that a template instance creates
+func (d *TemplateInstanceDescriber) DescribeObjects(objects []templateapi.TemplateInstanceObject, out *tabwriter.Writer) {
+	formatString(out, "Objects", " ")
+	indent := "    "
+	for _, o := range objects {
+		formatString(out, indent+o.Ref.Kind, fmt.Sprintf("%s/%s", o.Ref.Namespace, o.Ref.Name))
+	}
+}
+
+// DescribeSecret prints out information about the secret that holds the template instance parameters
+// kinternalprinter.SecretDescriber#Describe could have been used here, but the formatting
+// is off when it prints the information and seems to not be easily fixable
+func (d *TemplateInstanceDescriber) DescribeSecret(namespace, name string, out *tabwriter.Writer) {
+	c := d.kubeClient.Core().Secrets(namespace)
+	secret, err := c.Get(name, metav1.GetOptions{})
+
+	formatString(out, "Secret", " ")
+
+	if err != nil {
+		fmt.Fprintf(out, "%v", err)
+		return
+	}
```

Comment:
you need to consider what type of error you get.  If it's permission denied then you should just not print the parameters.  We are expecting someone might try to describe a templateinstance while not having permissions to view the secret that holds the parameters and if they do so, that's not a catastrophic error.

@jim-minter that does raise the question for me of, can a user create a templateinstance that refs a secret they cannot themselves view, and thereby force the controller to access the secret on their behalf?  Or do we have SAR checking against that?

Reply:
Updated to single out specific errors